src/metabase/query_processor/streaming/common.clj

```diff
@@ -16,24 +29,28 @@
   Object
   (format-value [this] this)
 
-  java.time.temporal.Temporal
-  (format-value [this]
-    (u.date/format this))
+  LocalDate
```

Comment:
i kinda wish this protocol was renamed from the very generic `format-value` to recognize its all about dates. And then the results timezone id can be got once and passed down through the signature each call rather than cached and held. This also makes the tests much nicer. I always get annoyed when i do something and get told i don't have an initialized qp.store, and therefore need to put a very data forward test inside of a with-database and with-everything-store macros.

That said, we're 20 minutes before a release so not something that needs to be done now but feels like the right call.

Reply:
You actually still need the QP store anyway in this case, since the `report-timezone` that gets used is dependent on the database from the QP store.  

We only override `format-value` for temporal types, but we still call it for *every* value when writing CSV/JSON -- the default impl for `Object` is just identity. So `format-temporal-value` or whatever I think would be a bit of a misnomer because we'd still be making calls like `(format-temporal-value 100.0)`